pkg/storegateway/chunkscache/cache.go

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package chunkscache
+
+import (
+	"context"
+
+	"github.com/oklog/ulid"
+	"github.com/prometheus/prometheus/tsdb/chunks"
+
+	"github.com/grafana/mimir/pkg/util/pool"
+)
+
+type Range struct {
+	BlockID   ulid.ULID
+	Start     chunks.ChunkRef
+	NumChunks int // We are certain about the number of chunks, not about their total size
+}
+
+type ChunksCache interface {
```

Comment:
If `ranges` is always required and `bytesPool` is sometimes `nil` my preference would be to make `bytesPool` the last argument

Reply:
i agree. previously ranges was variadic, so i couldn't do that. But when i changed it to a regular parameter, i didn't change the order too. refactored